packages/devtools_server/lib/src/handlers.dart

```diff
@@ -36,10 +36,50 @@ Future<shelf.Handler> defaultHandler() async {
 
   // Make a handler that delegates based on path.
   return (shelf.Request request) {
+    // The API handler takes all calls to api/.
+    if (Api.canHandle(request)) {
+      return Api.handle(request);
+    }
     return request.url.path.startsWith('packages/')
         // request.change here will strip the `packages` prefix from the path
         // so it's relative to packHandler's root.
         ? packHandler(request.change(path: 'packages'))
         : buildHandler(request);
   };
 }
+
+/// The DevTools server API.
+///
+/// This defines endpoints that serve all requests that come in over api/.
+class Api {
```

Comment:
Api is a bit too generic a class name. Absolutely anything can be an API.

What is this an API for?
Some bad Java ish name ideas: 
DevToolsServerDependencyInjector
ShelfHandler?
DevToolsServerHandler?

Reply:
The purpose of this class is to define the http API exposed by the server.  That is, any request that comes in as a call to `/api/` should be handled by a method on this class.

The key distinction is the separation of
* file handlers that only serve files, versus
* the API handler that gives business-logic-level answers to requests from the devtools frontend.

I can't think of another name that explains this distinction concisely, but suggestions are welcome.

Reply:
Maybe DevToolsHttpApi, DevToolsServerHttpApi, ServerHttpApi, DevToolsServerApi?

Reply:
How should we unify this with the SSE based Client-Server API that @DanTup is working on landing? It is alright to have both an HTTP and an SSE (websocket like API) but it does seem like overkill. One thing to keep in mind is the raw HTTP API like this could have security issues while the SSE one should be able to use an auth token to prevent calls from unauthorized users.
https://github.com/flutter/devtools/pull/987
I'm alright with any of the names Kenzie suggested although one of the shorter ones would be nice so it doesn't feel like we are writing enterprise java apps. :)

Reply:
Naming: I'd like to go with ServerApi.  DevToolsServerApi is nice, but redundant since we're in package:devtools.

SSE: We can discuss in-person tomorrow, but I suspect that SSE and HTTP are intended for reverse purposes and it's OK to have both.  We can also use secure tokens for HTTP requests to the server.